client/src/api/routes/applications.ts

```diff
@@ -0,0 +1,234 @@
+            cardColor,
+            progress
+        };
+        const { data, status } = await fetcher.patch<Partial<ApplicationObject>, AxiosResponse<ApplicationObject | Failure>>(`/application/${applicationId}`, body);
+        if(status !== 200) throw new Error(`${(data as Failure).message}\n\n${(data as Failure).error}`);
+        return (data as ApplicationObject);
+    });
+};
+
+/**
+ * @description PATCH /application/event/:applicationId/:eventId
+ * @param {string} applicationId 
+ * @param {string} eventId 
+ * @param {string} status 
+ * @param {string} title 
+ * @param {string} date 
+ * @param {string} location 
+ * @returns {UseQueryResult<EventObject>} the updated event
+ * @throws if fails
+ */
+export const useUpdateEvent = (applicationId: string, eventId: string, status?: boolean, title?: string, date?: string, location?: string): UseQueryResult<EventObject> => {
+    return useQuery('updateEvent', async () => {
+        const body: Partial<EventObject> = {
+            status,
+            title,
+            date,
+            location
+        };
+        const { data, status:statusCode } = await fetcher.patch<Partial<EventObject>, AxiosResponse<EventObject | Failure>>(`/application/event/${applicationId}/${eventId}`, body);
+        if(statusCode !== 200) throw new Error(`${(data as Failure).message}\n\n${(data as Failure).error}`);
+        return (data as EventObject);
+    });
+};
+
+/**
+ * @description PATCH /application/contact/:applicationID/:contactId
+ * @param {string} applicationId 
+ * @param {string} contactId 
+ * @param {string} name 
+ * @param {string} pronouns 
+ * @param {string} location 
+ * @param {string} phone 
+ * @param {string} email 
+ * @returns {UseQueryResult<ContactObject>} the updated contact
+ * @throws if fails
+ */
+export const useUpdateContact = (applicationId: string, contactId: string, name?: string, pronouns?: string, location?: string, phone?: string, email?: string): UseQueryResult<ContactObject> => {
+    return useQuery('updateContact', async () => {
+        const body: Partial<ContactObject> = {
+            name,
+            pronouns,
+            location,
+            phone,
+            email
+        };
+        const { data, status } = await fetcher.patch<Partial<ContactObject>, AxiosResponse<ContactObject | Failure>>(`/application/contact/${applicationId}/${contactId}`, body);
+        if(status !== 200) throw new Error(`${(data as Failure).message}\n\n${(data as Failure).error}`);
+        return (data as ContactObject);
+    });
+};
+
+/**
+ * @description DELETE /application/:applicationId
+ * @param {string} applicationId 
+ * @returns {UseQueryResult<boolean>} true if successful
+ * @throws if fails
+ */
+export const useDeleteApplication = (applicationId: string): UseQueryResult<boolean> => {
+    return useQuery('deleteApplication', async () => {
+        const { data, status } = await fetcher.delete<Failure>(`/application/${applicationId}`);
```

Comment:
Should these fetcher.delete functions has an alternative typing other than failure like the functions above?

Reply:
nah they only send a message on success so the Failure type covers  it. Can see why you are confused, but it works 👍